schemas/v1.0/azure.yaml.json

```diff
@@ -29,6 +29,13 @@
                     "examples": [
                         "todo-nodejs-mongo@0.0.1-beta"
                     ]
+                },
+                "minSupportedVersion": {
+                    "type": "string",
+                    "title": "The minimum version of azd needed for this project. Optional (allows all versions if absent).",
+                    "examples": [
+                        "0.6.0-beta.3"
+                    ]
```

Comment:
having a `string` seems a little fragile...  Should we define a type with Major, Minor, Patch and Beta as Integers?  
I imagine something like
```
- minSupportedVersion
  - Major: 0
  - Minor: 6
  - Patch: 0
  - Beta: 3
```

Reply:
There are many packages out there that can parse version numbers.  We shouldn't need a super verbose configuration API.  I would recommend we take an approach similar to terraform and support version constraints.

https://developer.hashicorp.com/terraform/language/expressions/version-constraints